### ISSUE TITLE
Pipelines can't build multi-module Maven projects

### ISSUE THREAD

**Issue by n-g-work** [epam/edp-tekton#5]
### Describe the bug

`Build` and `Code Review` pipelines with `Maven` fail for multi-module projects.
`Build` pipeline fails on building docker images. 
`Code Review` pipeline fails on `sonar` / `mvn-goals` task.

#### What did you do?

1. Create a sample multi-module `Java 8` / `Maven` project in EDP v2.12.1 Admin Console with the `Create` strategy. See screenshots section for details.
2. Clone the new repository and transfer it to EDP v3.2.1.
3. Make a change and start the `Code Review` pipeline.
4. Observe the failure on the Tekton `sonar` / `mvn-goals` task.
5. Ignore the failed `Code Review` and proceed to merge the change to start the `Build` pipeline.
6. Observe the failure on the Tekton `kaniko-build` / `build-and-push` task.

#### What did you expect to see?

A successfully executed `Code Review` and `Build` pipelines.

#### What did you see instead? Under which circumstances?

Failure on the `sonar` / `mvn-goals` task of the `Code Review` pipeline.
Failure on the `kaniko-build` / `build-and-push` task of the `Build` pipeline.

### Versions

EDP: **v3.2.1**
Git Provider: **Gerrit**
Kubernetes: **AWS EKS v1.24.11**

Tekton:
- Pipelines: **v0.45.0**
- Triggers: **v0.23.0**
- Trigger Interceptors: **v0.23.0**
  

### Screenshots

Create multi-module application in EDP v2.12.1:
![image](https://user-images.githubusercontent.com/101364051/234607957-88adea7b-aea9-407e-aaf1-05c7c84f4d27.png)

Failure on the Tekton `sonar` task of the `Code Review` pipeline:
![image](https://user-images.githubusercontent.com/101364051/234609079-41264639-150e-41da-8af7-f39f19e34545.png)
vvvvvvvvvvvvvvvvvv
![image](https://user-images.githubusercontent.com/101364051/234609763-6a85a33f-5e51-494f-b354-c0b4709c5530.png)

Failure on the Tekton `kaniko` task of the Build pipeline:
![image](https://user-images.githubusercontent.com/101364051/234610393-87278c61-e4b7-432f-b5bd-3c9708a22e4d.png)

### Additional context
 -

**Comment by n-g-work** [comment]
Updated: `Build` pipeline doesn't fail on the `sonar` task, only on the `kaniko`.

**Comment by NikolayMarusenko** [comment]
Hi @n-g-work!

Thanks for the detailed description, unfortunately in EDP: v3.2.1 in the Tekton deployment scenario there is currently no support for multi-module applications.

As a temporary solution, you can deploy the Jenkins scenario and create a kind: codebase through the yaml manifest (since it is not possible to indicate that this is a multi-module project through Headlamp UI).

Or apply the approach with customizing the Tekton pipelines according to this [case](https://epam.github.io/edp-install/use-cases/tekton-custom-pipelines/) and add the necessary functionality.

We have a planned task to migrate this functionality to the Tekton deployment scenario, we will begin the task as soon as possible.

**Comment by YevheniiPokhvalii** [comment]
Starting from the upcoming EDP 3.3 release, Tekton pipelines manage Java Maven Multi-Module projects out of the box the same way as the standard Java Maven projects. These pipelines are capable of recognizing Java deployable modules based on the information in the `pom.xml` file and performing relevant deployment actions. It's important to note that although the Dockerfile is typically located in the root directory, Kaniko uses the targets folder within the deployable module's context. For a clear illustration of a Multi-Module project structure, please refer to this [example](https://github.com/epmd-edp/java-maven-java17-multimodule.git).
Basically, the project structure should be the same as it was used for our Jenkins Maven Multi-Module pipeline.

Codebase example:
```
spec:
  buildTool: maven
  ciTool: tekton
  defaultBranch: main
  deploymentScript: helm-chart
  emptyProject: false
  framework: java11
  gitServer: gerrit
  gitUrlPath: /java-multimodule
  jenkinsSlave: null
  jiraIssueMetadataPayload: null
  jobProvisioning: null
  lang: java
  repository:
    url: https://github.com/epmd-edp/java-maven-java17-multimodule.git
  strategy: clone
  ticketNamePattern: null
  type: application
  versioning:
    type: default
```

Closes: #5
